tests/ftests/069-sudo-systemd_cgxget-cpu-settings-v1.py

```diff
@@ -0,0 +1,169 @@
+
+    ['cpu.cfs_quota_us',  '10000',       CGRP_VER_V1,
+     'cpu.cfs_quota_us',  '10000',       CGRP_VER_V1],
+    ['cpu.cfs_period_us', '100000',      CGRP_VER_V1,
+     'cpu.cfs_period_us', '100000',      CGRP_VER_V1],
+    ['cpu.cfs_period_us', '50000',       CGRP_VER_V1,
+     'cpu.max',           '10000 50000', CGRP_VER_V2],
+
+    ['cpu.cfs_quota_us',  '-1',         CGRP_VER_V1,
+     'cpu.cfs_quota_us',  '-1',         CGRP_VER_V1],
+    ['cpu.cfs_period_us', '100000',     CGRP_VER_V1,
+     'cpu.max',           'max 100000', CGRP_VER_V2],
+
+    ['cpu.max',           '5000 25000', CGRP_VER_V2,
+     'cpu.max',           '5000 25000', CGRP_VER_V2],
+    ['cpu.max',           '6000 26000', CGRP_VER_V2,
+     'cpu.cfs_quota_us',  '6000',       CGRP_VER_V1],
+    ['cpu.max',           '7000 27000', CGRP_VER_V2,
+     'cpu.cfs_period_us', '27000',      CGRP_VER_V1],
+
+    ['cpu.max',          'max 40000', CGRP_VER_V2,
+     'cpu.max',          'max 40000', CGRP_VER_V2],
+    ['cpu.max',          'max 41000', CGRP_VER_V2,
+     'cpu.cfs_quota_us', '-1',        CGRP_VER_V1],
+]
+
+
+def prereqs(config):
+    result = consts.TEST_PASSED
+    cause = None
+
+    if CgroupVersion.get_version('cpu') != CgroupVersion.CGROUP_V1:
+        result = consts.TEST_SKIPPED
+        cause = 'This test requires the cgroup v1 cpu controller'
+        return result, cause
+
+    if config.args.container:
+        result = consts.TEST_SKIPPED
+        cause = 'This test cannot be run within a container'
+
+    return result, cause
+
+
+def setup(config):
+    result = consts.TEST_PASSED
+    cause = None
+
+    Systemd.write_config_with_pid(config, CONFIG_FILE_NAME, SLICE, SCOPE)
+
+    Cgroup.configparser(config, load_file=CONFIG_FILE_NAME)
+
+    # create and check if the cgroup was created under the systemd default path
+    if not Systemd.create_and_validate(config, CONTROLLER, SYSTEMD_CGNAME):
+        result = consts.TEST_FAILED
+        cause = (
+                    'Failed to create systemd delegated cgroup {} under '
+                    '/sys/fs/cgroup/{}/{}/{}/'.format(SYSTEMD_CGNAME, CONTROLLER, SLICE, SCOPE)
+                )
+        return result, cause
+
+    # create and check if the cgroup was created under the controller sub-tree
+    if not Systemd.create_and_validate(config, CONTROLLER, OTHER_CGNAME, ignore_systemd=True):
+        result = consts.TEST_FAILED
+        cause = (
+                    'Failed to create cgroup {} under '
+                    '/sys/fs/cgroup/{}/'.format(OTHER_CGNAME, CONTROLLER)
+                )
+
+    return result, cause
+
+
+def test(config):
+    result = consts.TEST_PASSED
+    cause = None
+
+    cgrps = {SYSTEMD_CGNAME: False, OTHER_CGNAME: True}
+    for i in cgrps:
+        for entry in TABLE:
+            Cgroup.xset(config, cgname=i, setting=entry[0], value=entry[1],
+                        version=entry[2], ignore_systemd=cgrps[i])
```

Comment:
Dictionaries are only ordered on Python 3.7 and above.  Do we care that this code won't work on older versions of Python?  Should we enforce that?

Reply:
I'm specifically referring to the `cgrps[i]` snippet of code

Reply:
The unordered Dictionaries will not harm there, because `cgrps[SYSTEMD_CGNAME] = > ignore_systemd=False` and `cgrps[OTHER_CGNAME] => ignore_systemd=True` always no matter the order, and `xset` will find the right path of the cgroup.

Reply:
Thanks.  Makes sense